hack/.ci/run-e2e-gke.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2023 ScyllaDB
+#
+
+set -euExo pipefail
+shopt -s inherit_errexit
+
+if [ -z ${SO_IMAGE+x} ]; then
+  echo "SO_IMAGE can't be empty" > /dev/stderr
+  exit 2
+fi
+
+field_manager=hack
+
+function kubectl_create {
+    if [[ -z ${REENTRANT+x} ]]; then
+        # In an actual CI run we have to enforce that no two objects have the same name.
+        kubectl create --field-manager="${field_manager}" "$@"
+    else
+        # For development iterations we want to update the objects.
+        kubectl apply --server-side=true --field-manager="${field_manager}" --force-conflicts "$@"
+    fi
+}
+
+function gather-artifacts {
+  ARTIFACTS_DIR="${ARTIFACTS}" timeout -v 50m ./hack/ci-gather-artifacts.sh
+  # Events are costly to gather, so we'll erase them for now. (e2e namespaces are collected separately)
+  kubectl delete --raw /api/v1/namespaces/default/events || true
+  kubectl delete --raw /api/v1/namespaces/kube-system/events || true
+}
+
+function handle-exit {
+  gather-artifacts || "Error gathering artifacts" > /dev/stderr
+}
+
+trap handle-exit EXIT
+
+# Allow admin to use ephemeralcontainers
+kubectl_create -f - <<EOF
+  apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRole
+  metadata:
+    name: scylladb-e2e:hotfixes
+    labels:
+      rbac.authorization.k8s.io/aggregate-to-admin: "true"
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - pods/ephemeralcontainers
+    verbs:
+    - patch
+EOF
+
+kubectl apply --server-side -f ./pkg/api/scylla/v1alpha1/scylla.scylladb.com_nodeconfigs.yaml
+kubectl wait --for condition=established crd/nodeconfigs.scylla.scylladb.com
+
+kubectl_create -f - << EOF
+apiVersion: scylla.scylladb.com/v1alpha1
+kind: NodeConfig
+metadata:
+  name: cluster
+spec:
+  # TODO: make local disk setup create a disk img on a filesystem so we don't need the SSD.
+  localDiskSetup:
+    filesystems:
+    - device: /dev/nvme0n1
+      type: xfs
+    mounts:
+    - device: /dev/nvme0n1
+      mountPoint: /mnt/persistent-volumes
+      unsupportedOptions:
+      - prjquota
+  placement:
+    nodeSelector:
+      scylla.scylladb.com/node-type: scylla
+    tolerations:
+    - effect: NoSchedule
+      key: role
+      operator: Equal
+      value: scylla-clusters
+EOF
+
+kubectl_create -n local-csi-driver -f=https://raw.githubusercontent.com/scylladb/k8s-local-volume-provisioner/dcfceb52d9122192e50236463060cb5151d90dc4/deploy/kubernetes/local-csi-driver/{00_namespace,10_csidriver,10_driver_serviceaccount,10_provisioner_clusterrole,20_provisioner_clusterrolebinding,50_daemonset}.yaml
```

Comment:
Can't we just have a copy of manifests in `.ci` directory, as we do now with `actions`?

Reply:
ok, I'll add it, this was a snapshot from our own repo for the interim while it should eventually make it to the NodeConfig

Reply:
done